db/migrate/migration_utils/permission_adder.rb

```diff
@@ -32,17 +32,24 @@ module PermissionAdder
       module_function
 
       def add(having, add)
-        Role
+        role_scope = Role
           .joins(:role_permissions)
           .where(role_permissions: { permission: having.to_s })
           .references(:role_permissions)
-          .find_each do |role|
+
+        role_scope.find_each do |role|
           # Check if the add-permission already exists before adding
-          already_exists = RolePermission
-                             .exists?(role_id: role.id, permission: add.to_s)
-          unless already_exists
-            role.add_permission! add
-          end
+          next if RolePermission.exists?(role_id: role.id, permission: add.to_s)
+
+          added_permission = OpenProject::AccessControl.permission(add)
```

Comment:
Also nit: Shouldn't we move this assignment out of the loop, because `add` is not changed between iterations?

Reply:
It's a local lookup, so it does not really matter wher it's done but sure